### ISSUE TITLE
Remove deprecated twisted/python/timeoutqueue.py

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4551]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)| @thijstriemstra reported|
|-|-|
|Trac ID|trac#4551|
|Type|task|
|Created|2010-07-06 00:36:59Z|
|Branch|https://github.com/twisted/twisted/tree/remove-timeoutqueue-4551|

[twisted/python/timeoutqueue.py](twisted/python/timeoutqueue.py) has been deprecated since r20510 ([#2536](https://github.com/twisted/twisted/issues/2536)), time to remove it.

<details><summary>Searchable metadata</summary>

```
trac-id__4551 4551
type__task task
reporter__thijs thijs
priority__lowest lowest
milestone__ 
branch__branches_remove_timeoutqueue_4551 branches/remove-timeoutqueue-4551
branch_author__cyli cyli
status__closed closed
resolution__fixed fixed
component__core core
keywords__easy easy
time__1278376619000000 1278376619000000
changetime__1288126501000000 1288126501000000
version__None None
owner__ 
cc__thijs cc__jesstess
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_2"></a>|@cyli commented|
|-|-|

(In [30160]) Branching to 'remove-timeoutqueue-4551'

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_3"></a>|@cyli commented|
|-|-|

(In [30161]) Remove timeoutqueue and test_timeoutqueue.
Refs [#4551](https://github.com/twisted/twisted/issues/4551)

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_4"></a>|@cyli commented|
|-|-|

(In [30162]) Added a .removal topfile.
Refs [#4551](https://github.com/twisted/twisted/issues/4551)

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_5"></a>|@cyli removed owner|
|-|-|

Removed timeoutqeue and its tests.  There were no references to it - all tests pass.

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_6"></a>|@cyli commented|
|-|-|

[build results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/remove-timeoutqueue-4551)

**Comment by twisted-trac** [comment]
|[<img alt="jesstess's avatar" src="https://avatars.githubusercontent.com/u/188336?s=50" width="50" height="50">](https://github.com/jesstess)<a name="note_7"></a>|@jesstess set owner to @cyli|
|-|-|

Thanks for working on all these deprecations, cyli!

ReviewProcess suggests a construction like "twisted.internet.interfaces.IReactorTime.cancelCallLater, deprecated since Twisted 2.5, has been removed." for removal NEWS files, including when it was deprecated, so can you change the NEWS file to that format?

Other than that, this looks great and ready to merge.

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_8"></a>|@cyli commented|
|-|-|

(In [30177]) Changed newsfile to conform with the accepted format.

Refs [#4551](https://github.com/twisted/twisted/issues/4551)

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_9"></a>|@cyli commented|
|-|-|

Done, and thanks for the review!  Merging.

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_10"></a>|@cyli set status to `closed`|
|-|-|

(In [30178]) Merge remove-timeoutqueue-4551: Removed deprecated twisted.python.timeoutqueue

Author: cyli
Reviewer: jesstess
Fixes: [#4551](https://github.com/twisted/twisted/issues/4551)

twisted.python.timeoutqueue has been deprecated since 2007.  Removed it and its tests.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
